iOS/Sources/Beagle/Sources/Analytics/Service/ActionAttributes.swift

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+import UIKit
+
+internal enum ActionAttributes: Equatable {
+    case all
+    case some([String])
+}
+
+extension Action {
+
+    func getAttributes(_ attributes: ActionAttributes, contextProvider: UIView) -> [String: DynamicObject] {
+        if case .some(let array) = attributes, array.isEmpty { return [:] }
+
+        let dynamicObject = transformToDynamicObject(self)
+            .evaluate(with: contextProvider)
+            // don't use analytics attribute
+            .set(.empty, with: analyticsPath)
+
+        let dict = dynamicObject.asDictionary()
+
+        switch attributes {
+        case .some(let some):
+            return dict.getSomeAttributes(some, contextProvider: contextProvider)
+
+        case .all:
+            return dict
+        }
+    }
+}
+
+private func pathForAttribute(_ attribute: String) -> Path? {
+    guard let path = Path(rawValue: attribute) else { return nil }
+
+    for node in path.nodes {
+        if case .index = node { return nil }
+    }
+    return path
+}
+
+extension Dictionary where Key == String, Value == DynamicObject {
```

Comment:
May create a type alias to easily define this extension and also can be use in other places on the application.
```suggestion
public typealias DynamicObjectDictionary = [String: DynamicObject]
extension DynamicObjectDictionary {
```